components/script/dom/webgl2renderingcontext.rs

```diff
@@ -445,7 +730,23 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
 
     /// https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
     fn DeleteBuffer(&self, buffer: Option<&WebGLBuffer>) {
-        self.base.DeleteBuffer(buffer)
+        let buffer = match buffer {
+            Some(buffer) => buffer,
+            None => return,
+        };
+        handle_potential_webgl_error!(self.base, self.base.validate_ownership(buffer), return);
+        if buffer.is_marked_for_deletion() {
+            return;
+        }
+        self.base.current_vao().unbind_buffer(buffer);
+        self.unbind_from(&self.base.array_buffer_slot(), &buffer);
+        self.unbind_from(&self.bound_copy_read_buffer, &buffer);
+        self.unbind_from(&self.bound_copy_write_buffer, &buffer);
+        self.unbind_from(&self.bound_pixel_pack_buffer, &buffer);
+        self.unbind_from(&self.bound_pixel_unpack_buffer, &buffer);
+        self.unbind_from(&self.bound_transform_feedback_buffer, &buffer);
+        self.unbind_from(&self.bound_uniform_buffer, &buffer);
+        buffer.mark_for_deletion(false);
```

Comment:
Why all this new stuff?

Reply:
The WebGL 1 DeleteBuffer implementation deals with less buffers compared to WebGL 2. Do you have some ideas for improving this code?

Reply:
Nope, that seems like a pretty straightforward way to do that.